tests/unit/contracts/test_layers.py

```diff
@@ -138,38 +142,44 @@ def test_illegal_imports(self, specify_container: bool):
 
         contract_check = contract.check(graph=graph, verbose=False)
 
-        assert contract_check.kept is False
-        sorted_metadata = _get_sorted_metadata(contract_check)
-        assert sorted_metadata == {
-            "invalid_dependencies": [
-                {
-                    "importer": "mypackage.medium_a",
-                    "imported": "mypackage.medium_b",
-                    "routes": [
-                        {
-                            "chain": [
-                                {
-                                    "importer": "mypackage.medium_a.blue",
-                                    "imported": "mypackage.medium_b.red",
-                                    "line_numbers": (3,),
-                                },
-                            ],
-                            "extra_firsts": [],
-                            "extra_lasts": [],
-                        }
-                    ],
-                },
-            ],
-            "undeclared_modules": set(),
-        }
+        if siblings_independent:
+            assert contract_check.kept is False
+            sorted_metadata = _get_sorted_metadata(contract_check)
+            assert sorted_metadata == {
+                "invalid_dependencies": [
+                    {
+                        "importer": "mypackage.medium_a",
+                        "imported": "mypackage.medium_b",
+                        "routes": [
+                            {
+                                "chain": [
+                                    {
+                                        "importer": "mypackage.medium_a.blue",
+                                        "imported": "mypackage.medium_b.red",
+                                        "line_numbers": (3,),
+                                    },
+                                ],
+                                "extra_firsts": [],
+                                "extra_lasts": [],
+                            }
+                        ],
+                    },
+                ],
+                "undeclared_modules": set(),
+            }
+        else:
+            sorted_metadata = _get_sorted_metadata(contract_check)
+            assert contract_check.kept is True
```

Comment:
Maybe this test is doing too much? For a start it is called `test_illegal_imports`, but one of the cases is for no illegal imports.

I think I would lean towards having an entirely separate test case, or at least renaming the test to be accurate. Thoughts?

Reply:
I quite like to use the same test so that we can sure the setup is the same, so that we can be sure that there is no other unintended difference between the test cases. For now I've renamed the test to `test_imports_between_sibling_modules`.

Reply:
Ok, no problem!